contracts/utils/cryptography/Hashes.sol

```diff
@@ -0,0 +1,27 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.0;
+
+/**
+ * @dev Library of standard hash functions.
+ */
+library Hashes {
+    /**
+     * @dev Keccak256 hash of a sorted pair of bytes32. Frequently used when working with merkle proofs.
+     */
+    function stdPairHash(bytes32 a, bytes32 b) internal pure returns (bytes32) {
+        return a < b ? _efficientHash(a, b) : _efficientHash(b, a);
```

Comment:
Just realized this function is commutative. Should the tree work with non-commutative functions?

I guess it should, so I'd feel better if we add tests with a custom non-commutative `keccak256` function. It may require updating the javascript library for testing.

If the tree expects only commutative functions we should be explicit about it.

Reply:
We should support non commutative hashes, so yes, we should test that.

Reply:
Right, we should do this after the javascript library is released.